tests/queries/test_qs_combinators.py

```diff
@@ -257,6 +257,23 @@ def test_union_with_two_annotated_values_list(self):
         )
         self.assertCountEqual(qs1.union(qs2), [(1, 0), (1, 2)])
 
+    def test_union_with_field_and_annotation_values(self):
+        qs1 = (
+            Number.objects.filter(num=1)
+            .annotate(
+                zero=Value(0, IntegerField()),
+            )
+            .values_list("num", "zero")
+        )
+        qs2 = (
+            Number.objects.filter(num=2)
+            .annotate(
+                zero=Value(0, IntegerField()),
+            )
+            .values_list("zero", "num")
+        )
+        self.assertCountEqual(qs1.union(qs2), [(1, 0), (0, 2)])
```

Comment:
This use to fail because before it was doing

```sql
SELECT num, 0 AS zero
FROM number
UNION
SELECT num, 0 AS zero
FROM number
```

Completely disregarding the explicit order specified to `values_list`